php/proxy.php

```diff
@@ -3,150 +3,192 @@
+            $this->settings = self::getSettings();
+            $this->username = 'username:'.$this->settings['username'];
+            $this->password = 'password:'.$this->settings['password'];
+            array_push($headers, $this->username);
+            array_push($headers, $this->password);
+            array_push($headers,'User-Agent:Drupal Proxy');
+
         }
         return $headers;
     }
-    
+
     private function getPOSTdata(){
         return file_get_contents("php://input");
     }
-    
-    private function proxyRequest($url){
+
+    public function __construct() {
+        self::disable_gzip();
+        $this->settings = self::getSettings();
+        $this->targetPath = rtrim($this->settings['url'],"/");
+        $this->host = parse_url($this->targetPath)['host'];
+
+        if(isset($_SERVER['HTTPS'] ) && $_SERVER['HTTPS'] == 'on'){
+            $this->domainURL = 'https://'.$_SERVER['HTTP_HOST'];
+        } else {
+            $this->domainURL = 'http://'.$_SERVER['HTTP_HOST'];
+        }
+
+        $part = '/acrolinx/proxy.php'; //TODO: Ensure that this is the correct location of the 'proxy.php' on the server.
+        $requestURI = $_SERVER['REQUEST_URI'];
+        $pos = self::getPosition($requestURI, $part);
+
+        if ($pos !== false) {
+            $requiredURI = self::getActualURI($requestURI, $pos, $part);
+            $this->url = $this->targetPath.$requiredURI;
+        }
+    }
+
+    public function processRequest() {
         $ch = curl_init();
-        curl_setopt ($ch, CURLOPT_URL, $url);
-//To write all requests to std error, uncomment the following line of code.
-//By default, Apache will write to /var/log/apache2/error.log.
-//Don’t use this sample in production. All passwords, cookies, and tokens are logged.
-//	    curl_setopt ($ch, CURLOPT_VERBOSE, true); 
-        
+        curl_setopt ($ch, CURLOPT_URL, $this->url);
+
+         //To write all requests to std error, uncomment the following line of code.
+         //By default, Apache will write to /var/log/apache2/error.log.
+         //Don't use this sample in production. All passwords, cookies, and tokens are logged.
+         //curl_setopt ($ch, CURLOPT_VERBOSE, true);
+
         if($_SERVER['REQUEST_METHOD'] == 'POST'){
-            $postData = self::getPOSTdata();  
+            $postData = self::getPOSTdata();
             curl_setopt ($ch, CURLOPT_POST, 1);
             curl_setopt ($ch, CURLOPT_POSTFIELDS, $postData);
         }
-        
+
         if($_SERVER['REQUEST_METHOD'] == 'PUT'){
-            curl_setopt($ch, CURLOPT_PUT, 1);
+            $postData = self::getPOSTdata();
+            curl_setopt($ch, CURLOPT_HTTPHEADER, array('Content-Type: ' . $_SERVER["CONTENT_TYPE"]  . ''));
+            curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'PUT');
+            curl_setopt($ch, CURLOPT_POSTFIELDS,$postData);
         }
-        
-        $headers = self::getHeadersOfRequest(); 
-     
+
+        if($_SERVER['REQUEST_METHOD'] == 'DELETE'){
```

Comment:
What do we do with unhandled methods?

Reply:
Fix: Return error status code 405 (Method Not Allowed)